blaze/expr/collections.py

```diff
@@ -418,6 +418,7 @@ def on_right(self):
         return on_right
 
     @property
+    @memoize
     def schema(self):
```

Comment:
@llllllllll sorry for the delay getting back to you -- yes, agreed on the descriptor for `schema`.  Thanks for the example code, my attempt at this consistently ran into a recursion limit exception.  I'll put in your version and test it out.